### PR TITLE
Added module for deploying Cognito user pool

### PR DESCRIPTION
**Pull request by RustamGimadiev**
A part of ArgoCD/Kubeflow changes

### CODE REVIEW

File: modules/cognito/main.tf [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+resource aws_route53_record this {
+  name    = aws_cognito_user_pool_domain.this.domain
+  type    = "A"
+  zone_id = var.zone_id
+  alias {
+    evaluate_target_health = false
+    name                   = aws_cognito_user_pool_domain.this.cloudfront_distribution_arn
+    zone_id                = "Z2FDTNDATAQYW2"
```
Comment (akastav) [inline modules/cognito/main.tf:18]: hardcode?

File: modules/cognito/main.tf [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  providers = {
+    aws = aws.cognito
+  }
+
+  tags = var.tags
+}
+
+provider aws {
+  alias  = "cognito"
+  region = "us-east-1"
```
Comment (akastav) [inline modules/cognito/main.tf:47]: hardcode

**Review by akastav — CHANGES_REQUESTED** [review]

File: modules/cognito/main.tf (hunk 1 above)
Comment (RustamGimadiev) [reply modules/cognito/main.tf:20]: Z2FDTNDATAQYW2 is always the hosted zone ID when you create an alias record that routes traffic to a CloudFront distribution.

File: modules/cognito/main.tf (hunk 2 above)
Comment (RustamGimadiev) [reply modules/cognito/main.tf:52]: AWS Cognito uses the N.Virginia region to deploying its own CDN system, that a reason why we should create ACM certificates it that zone

**Review by akastav — APPROVED** [review]

**Pull request merged**